hishel/_async/_transports.py

```diff
@@ -245,11 +245,13 @@ async def _create_hishel_response(
         metadata: Metadata | None = None,
     ) -> Response:
         if cached:
-            assert metadata
-            metadata["number_of_uses"] += 1
-            await self._storage.update_metadata(key=key, request=request, response=response, metadata=metadata)
-            response.extensions["from_cache"] = True  # type: ignore[index]
-            response.extensions["cache_metadata"] = metadata  # type: ignore[index]
+            if self._controller._update_metadata_on_hit or revalidated:
```

Comment:
Why do we check `revalidated` here? It seems like we should just do `self._controller._update_metadata_on_hit`